lib/node.js

```diff
@@ -822,8 +822,102 @@ declare module "vm" {
+  declare var Z_MAX_MEMLEVEL: number;
+  declare var Z_MAX_WINDOWBITS: number;
+  declare var Z_MIN_CHUNK: number;
+  declare var Z_MIN_LEVEL: number;
+  declare var Z_MIN_MEMLEVEL: number;
+  declare var Z_MIN_WINDOWBITS: number;
+  declare var codes: {
+    Z_OK: number,
+    Z_STREAM_END: number,
+    Z_NEED_DICT: number,
+    Z_ERRNO: number,
+    Z_STREAM_ERROR: number,
+    Z_DATA_ERROR: number,
+    Z_MEM_ERROR: number,
+    Z_BUF_ERROR: number,
+    Z_VERSION_ERROR: number
+  };
+  declare class Zlib {
+    // TODO
+  }
+  declare class Deflate extends Zlib {}
+  declare class Inflate extends Zlib {}
+  declare class Gzip extends Zlib {}
+  declare class Gunzip extends Zlib {}
+  declare class DeflateRaw extends Zlib {}
+  declare class InflateRaw extends Zlib {}
+  declare class Unzip extends Zlib {}
+  declare function createDeflate(options?: zlib$ZlibOptions): Deflate;
+  declare function createInflate(options?: zlib$ZlibOptions): Inflate;
+  declare function createDeflateRaw(options?: zlib$ZlibOptions): DeflateRaw;
+  declare function createInflateRaw(options?: zlib$ZlibOptions): InflateRaw;
+  declare function createGzip(options?: zlib$ZlibOptions): Gzip;
+  declare function createGunzip(options?: zlib$ZlibOptions): Gunzip;
+  declare function createUnzip(options?: zlib$ZlibOptions): Unzip;
+  declare function deflate(buffer: string | Buffer, options: zlib$ZlibOptions, callback?: ((error: ?Error, ...args: Array<any>) => void)): void;
+  declare function deflateSync(buffer: string | Buffer, options: zlib$ZlibOptions): void;
+  declare function gzip(buffer: string | Buffer, options: zlib$ZlibOptions, callback?: ((error: ?Error, ...args: Array<any>) => void)): void;
+  declare function gzipSync(buffer: string | Buffer, options: zlib$ZlibOptions): void;
+  declare function deflateRaw(buffer: string | Buffer, options: zlib$ZlibOptions, callback?: ((error: ?Error, ...args: Array<any>) => void)): void;
+  declare function deflateRawSync(buffer: string | Buffer, options: zlib$ZlibOptions): void;
+  declare function unzip(buffer: string | Buffer, options: zlib$ZlibOptions, callback?: ((error: ?Error, ...args: Array<any>) => void)): void;
+  declare function unzipSync(buffer: string | Buffer, options: zlib$ZlibOptions): void;
+  declare function inflate(buffer: string | Buffer, options: zlib$ZlibOptions, callback?: ((error: ?Error, ...args: Array<any>) => void)): void;
+  declare function inflateSync(buffer: string | Buffer, options: zlib$ZlibOptions): void;
+  declare function gunzip(buffer: string | Buffer, options: zlib$ZlibOptions, callback?: ((error: ?Error, ...args: Array<any>) => void)): void;
+  declare function gunzipSync(buffer: string | Buffer, options: zlib$ZlibOptions): void;
+  declare function inflateRaw(buffer: string | Buffer, options: zlib$ZlibOptions, callback?: ((error: ?Error, ...args: Array<any>) => void)): void;
+  declare function inflateRawSync(buffer: string | Buffer, options: zlib$ZlibOptions): void;
```

Comment:
You could avoid the duplication in the above by defining `zlib$asyncFn` and `zlib$syncFn` types and declare these as vars with those function type—which would be equivalent to `declare function`, iiuc.

The option argument to these functions is optional in the docs, but not market optional here.

I think you can be even more specific with the callback function types, too. `(error: ?Error, result: any) => void` should do it.